### PR TITLE
i#1698 ldstex: Add exclusive memop create and query support

### PR DESCRIPTION
**Pull request by derekbruening**
Adds a new API function instr_is_exclusive_load().  Moves the existing
private implementation (for -unsafe_build_ldstex) to become a public
function on AArch64.

Adds new instruction creation macros for exclusive load, store, and
clear opcodes.

Adds tests to api.ir for instr_is_exclusive_* and the new creation
macros.  This necessitated switching api.ir to use runcmp.cmake, since
its output is now too large for a CMake regular expression match.

Issue: #1698

### CODE REVIEW

**Comment by derekbruening** [comment]
OK trying this chained PR again vs the PR #4543.

**Comment by derekbruening** [comment]
Hmm the diff has some of the base in it.  Maybe if I had merged the fix a different way.  Github doesn't have great support for chained PR's.  In any case it's still quite reviewable IMHO; will update and ensure green before merge of course.  Let me know if it's not easy to look at.

**Comment by derekbruening** [comment]
@abhinav92003 if you could take this one over from Assad as well.  #4545 is chained on top of this so this needs to go in before it.

File: suite/tests/api/ir_aarch64.c [hunk 1]
```diff
@@ -4342,6 +4342,84 @@ test_sys(void *dc)
...
+                               OPND_CREATE_MEM8(DR_REG_X1, 0));
+    ASSERT(instr_is_exclusive_load(instr));
+    test_instr_encoding(dc, OP_ldxrb, instr);
+    instr = INSTR_CREATE_ldxrh(dc, opnd_create_reg(DR_REG_W0),
+                               OPND_CREATE_MEM16(DR_REG_X1, 0));
+    ASSERT(instr_is_exclusive_load(instr));
+    test_instr_encoding(dc, OP_ldxrh, instr);
+    instr = INSTR_CREATE_ldxp(dc, opnd_create_reg(DR_REG_W0), opnd_create_reg(DR_REG_W1),
+                              OPND_CREATE_MEM64(DR_REG_X2, 0));
+    ASSERT(instr_is_exclusive_load(instr));
```
Comment (abhinav92003) [inline suite/tests/api/ir_aarch64.c:4364]: Perhaps also test the 64-bit version of `ldxp` and `stxp`?

**Review by abhinav92003 — APPROVED** [review]

File: suite/tests/api/ir_aarch64.c (hunk 1 above)
Comment (derekbruening) [reply suite/tests/api/ir_aarch64.c:4364]: Done.

**Comment by derekbruening** [comment]
Ah, the tests are failing on clrex, which had a fix (it has a fake source!  #4532 covers removing) that got separated when I made this separate branch for this PR separate from the main #1698 tree.  Fixed now.

**Pull request merged**